python/ray/data/block.py

```diff
@@ -21,6 +21,43 @@
 KeyType = TypeVar("KeyType")
 AggType = TypeVar("AggType")
 
+# A function that extracts a concrete value from a record in a Dataset, used
+# in ``sort(value_fns...)``, ``groupby(value_fn).agg(Agg(value_fn), ...)``.
+# It can either be None (intepreted as the identity function), the name
+# of a Dataset column, or a lambda function that extracts the desired value
+# from the object.
+KeyFn = Union[None, str, Callable[[T], Any]]
```

Comment:
Hmm, it's not always a `Fn`, should we call it KeyT or KeyType?

Reply:
Even if it's a string semantically it's implementing a function (mapping records -> value).